crates/typst/src/foundations/calc.rs

```diff
@@ -908,6 +910,82 @@ pub fn quo(
     Ok(floor(dividend.apply2(divisor.v, Div::div, Div::div)))
 }
 
+/// Calculates the p-norm of a sequence of values.
+///
+/// ```example
+/// #calc.norm(1, 2, -3, 0.5)
+/// #calc.norm(p: 3, 1in, 2cm)
+/// #calc.norm(3em, 4em)
+/// ```
+#[func(title = "P-Norm")]
+pub fn norm(
+    /// The p value to calculate the p-norm of.
+    #[named]
+    #[default(Num::Int(2))]
+    p: Num,
+    /// The sequence of values from which to calculate the p-norm.
+    /// Returns `0.0` if empty.
+    #[variadic]
+    values: Vec<Spanned<Value>>,
+) -> SourceResult<Value> {
+    let mut sum = 0.0;
+    if let Some(Spanned { v, span }) = values.first() {
```

Comment:
Flipping this `if let` into a `let .. else ` would make this a bit eaiser to read because of reduced indentation.